explorer/BUILD

```diff
@@ -42,8 +43,72 @@ cc_binary(
     ],
 )
 
-cc_library(
-    name = "file_test_common",
+EXPLORER_GTEST_TESTS = [
+    (
+        "ast",
+        "ast_test_matchers_test",
+        "ASTTestMatchers",
+    ),
+    (
+        "ast",
+        "expression_test",
+        "ExpressionTest",
+    ),
+    (
+        "ast",
+        "element_test",
+        "ElementTest",
+    ),
+    (
+        "ast",
+        "pattern_test",
+        "PatternTest",
+    ),
+    (
+        "base",
+        "arena_test",
+        "ArenaTest",
+    ),
+    (
+        "base",
+        "decompose_test",
+        "HashFromDecomposeTest",
+    ),
+    (
+        "base",
+        "error_builders_test",
+        "ErrorBuildersTest",
+    ),
+    (
+        "base",
+        "set_file_context_raii_test",
+        "SetFileContextRaiiTest",
+    ),
+    (
+        "base",
+        "set_program_phase_raii_test",
+        "SetProgramPhaseRaiiTest",
+    ),
+    (
+        "parse_and_execute",
+        "parse_and_execute_test",
+        "ParseAndExecuteTest",
+    ),
+    (
+        "syntax",
+        "parse_test",
+        "ParseTest",
+    ),
+    (
+        "syntax",
+        "unimplemented_example_test",
+        "UnimplementedExampleTest",
+    ),
```

Comment:
Versus the --gtest_filter approach, had you considered replacing the built-in googletest main with something that'd dispatch based on the target name? That'd remove the need to add test names to the list, it could just do direct dependencies.

e.g., with a registration hook like `if (getenv(TEST_TARGET) == my_target && absl::GetFlag(gtest_filter).empty()) { absl::SetFlag(gtest_filter, my_tests); }` to set within the binary, rather than in the BUILD. My thought there is keeping the test unit names next to the code.

If you combined that with a .bzl rule that combined the per-cc_library with a sh_test that depended back on the main busy box binary, you might even be able to place such a list straight into `deps`.

Reply:
(Comment no longer applies to latest version of this PR.)